hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/converter/FSConfigToCSConfigConverter.java

```diff
@@ -310,6 +313,7 @@ private void convertCapacitySchedulerXml(FairScheduler fs) {
         .withPercentages(usePercentages)
         .build();
 
+    queueConverter.emitDefaultUserLimitFactor(fsLeafQueue, capacitySchedulerConfig);
```

Comment:
This should ideally be called from inside the queueConverter, as this method operates on queues themselves, not on the whole structure. This way there is no need to get the leafQueues array.